lib/tasks_visitor.rb

```diff
@@ -15,17 +15,19 @@ def noop(n, p)
     n
   end
 
-  def visit_hash(n, path)
+  def visit_Hash(n, path)
     n.each do |k, v|
       n[k] = visit(v, path + [k], k)
     end
   end
+  
+  alias :visit_ActiveSupport_HashWithIndifferentAccess :visit_Hash
```

Comment:
That should be an exhaustive list of types its possible to get back from
json.
On Jan 21, 2015 4:53 AM, "Campbell Allen" <notifications@github.com> wrote:

> In lib/tasks_visitor.rb
> <https://github.com/zooniverse/Panoptes/pull/441#discussion_r23291216>:
>
> >      n.each do |k, v|
> >        n[k] = visit(v, path + [k], k)
> >      end
> >    end
> > +
> > +  alias :visit_ActiveSupport_HashWithIndifferentAccess :visit_Hash
>
> Is this alias list exhaustive of the expected types? If not what about
> using method_missing to dispatch the relevant class method?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zooniverse/Panoptes/pull/441/files#r23291216>.
>